db/view/view.cc

```diff
@@ -318,6 +320,50 @@ static bool is_partition_key_empty(
     }
 }
 
+// Checks if the result matches the provided view filter
+class view_filter_checking_visitor {
+    const schema& _base;
+    const view_info& _view;
+    const std::vector<bytes>& _pk;
+    const std::vector<bytes>& _ck;
+
+    bool _matches_view_filter = true;
+public:
+    view_filter_checking_visitor(const schema& base, const view_info& view, const std::vector<bytes>& pk, const std::vector<bytes>& ck)
+        : _base(base)
+        , _view(view)
+        , _pk(pk)
+        , _ck(ck)
+    {}
+
+    void accept_new_partition(const partition_key& key, uint64_t row_count) {}
+    void accept_new_partition(uint64_t row_count) {}
+
+    void accept_new_row(const clustering_key& key, const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter &= check_if_matches(static_row, row);
```

Comment:
@psarna you said you don't believe anyone uses view filtering. But isn't IS NOT NULL mandatory?

Perhaps it's implicit and/or stripped out. But logically, we can't create view rows when columns that become part of the view primary key are NULL.

Reply:
IS NOT NULL Is syntactic sugar (or rather syntactic bile™, since it uglifies the process of creating views), it doesn't evaluate to anything. But it **is** possible to create a filtering view like `create MATERIALIZED VIEW v AS SELECT * from t where v < 3 primary key (v,id);`, so we must support the filtering for backward compatibility if nothing else.